src/editor/codemirror/edits.ts

```diff
@@ -89,29 +93,12 @@ export const calculateChanges = (state: EditorState, addition: string) => {
     changes[changes.length - 1].insert += "\n\n";
   }
 
-  // We'll want to add more sophisticated insertion than this.
   if (addition) {
-    // We always want a newline at the end as for now our inserts are whole lines.
-    // If the insertion point is after a node then we want an newline before it.
-    // If the insertion point is before a node then we want an newline after it to break the line.
-    const lastImport = allCurrentImports?.[allCurrentImports.length - 1]?.node;
-    let relation = Relation.Default;
-    let insertionPoint = 0;
-    if (lastImport?.nextSibling) {
-      relation = Relation.Before;
-      insertionPoint = lastImport.nextSibling.from;
-    } else if (lastImport) {
-      relation = Relation.After;
-      insertionPoint = lastImport.to;
-    }
-
+    const line = state.doc.lineAt(pos!);
```

Comment:
We need to reinstate code that works when pos is undefined (as it is when clicking "Insert code"). I've dropped this as it wasn't line oriented and clearly needs to be (see #365). However, it's broken lots of tests, so either a quick fix or reinstate...